storage/statedb/cache_redis_test.go

```diff
@@ -52,11 +50,19 @@ func _TestSubscription(t *testing.T) {
 
 		ch := cache.SubscribeBlockCh()
 
-		actualMsg := <-ch
-		assert.Equal(t, msg1, actualMsg.Payload)
+		select {
+		case actualMsg := <-ch:
+			assert.Equal(t, msg1, actualMsg.Payload)
+		case <-time.After(time.Second):
+			panic("timeout")
```

Comment:
How about using `t.Fail()` or `t.FailNow()`?

Reply:
In this case, `Fail()` and `FailNow()` cannot stop the running test. Refer to the description of `FailNow()`.
```
FailNow must be called from the goroutine running the
test or benchmark function, not from other goroutines
created during the test. Calling FailNow does not stop
those other goroutines.
```